dll/win32/shlwapi/utils.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * PROJECT:     ReactOS Shell
+ * LICENSE:     LGPL-2.0-or-later (https://spdx.org/licenses/LGPL-2.0-or-later)
+ * PURPOSE:     Implement shell light-weight utility functions
+ * COPYRIGHT:   Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
+ */
+
+#define _ATL_NO_EXCEPTIONS
+#include "precomp.h"
+#include <shellapi.h>
+#include <shlwapi.h>
+#include <shlwapi_undoc.h>
+#include <shlobj_undoc.h>
+#include <shlguid_undoc.h>
+#include <atlstr.h>         // for CStringW
+#include <strsafe.h>        // for StringC... functions
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+/*************************************************************************
+ * IContextMenu_Invoke [SHLWAPI.207]
+ *
+ * Used by Win:SHELL32!CISFBand::_TrySimpleInvoke.
+ */
+EXTERN_C
+BOOL WINAPI
+IContextMenu_Invoke(
+    _In_ IContextMenu *pContextMenu,
+    _In_ HWND hwnd,
+    _In_ LPCSTR lpVerb,
+    _In_ UINT uFlags)
+{
+    CMINVOKECOMMANDINFO info;
+    BOOL ret = FALSE;
+    INT iDefItem = 0;
+    HMENU hMenu = NULL;
+    HCURSOR hOldCursor;
+
+    TRACE("(%p, %p, %s, %u)\n", pContextMenu, hwnd, debugstr_a(lpVerb), uFlags);
+
+    if (!pContextMenu)
+        return FALSE;
+
+    hOldCursor = SetCursor(LoadCursorW(NULL, (LPCWSTR)IDC_WAIT));
```

Comment:
It doesn't seem that the operations done between this call and the other restoring SetCursor call below take any substantial amount of code. Excepting maybe the code made only within the IS_INTRESOURCE if-case (and just for the QueryContextMenu call, maybe).
OR, maybe the InvokeComamnd call is the one that takes some time to run, and would benefit of a change of cursor.

Reply:
Moved `SetCursor(hOldCursor);`.